packages/frontend/core/src/modules/services.ts

```diff
@@ -23,5 +29,12 @@ export function configureBusinessServices(services: ServiceCollection) {
 }
 
 export function configureWebInfraServices(services: ServiceCollection) {
-  services.addImpl(GlobalCache, LocalStorageGlobalCache);
+  services
+    .addImpl(GlobalCache, LocalStorageGlobalCache)
+    .addImpl(GlobalState, LocalStorageGlobalState)
+    .scope(WorkspaceScope)
```

Comment:
is it better to be put in `configureWorkspaceServices`, because I saw an overriding of `WorkspaceLocalState` in testing environment, but I didn't see it in normal environment. It makes me confused at first until I search with it globally